src/waveform/widgets/glwaveformwidgetabstract.h

```diff
@@ -23,6 +23,12 @@ class GLWaveformWidgetAbstract : public WaveformWidgetAbstract, public WGLWavefo
     }
 
   protected:
+#ifdef MIXXX_USE_QOPENGL
+    void renderGL() override {
+        // called by OpenGLWindow to avoid flickering on resize
+        static_cast<WaveformWidgetAbstract*>(this)->render();
+    }
```

Comment:
The static cast seems unnecessary? I thought the correct way to call parent functions was `WaveformWidgetAbstract::render()`?

Reply:
WaveformWidgetAbstract::render() is a virtual. Calling it like that will call that exact function, in the base class, and not the override.

Reply:
Ah, I thought that was the intention. Then if virtual dispatch is desired, why would just calling plain `render()` without the cast not work?

Reply:
Because of ambiguity. QWidget also has a method render(...) and even though that one has arguments, the compiler doesn't like this.